From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Sun, 5 Oct 2014 17:01:31 +0200
Subject: Add Capabilities iterator

---
 swig/Capabilities.i | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/swig/Capabilities.i b/swig/Capabilities.i
index 88e5c30..3ec97f5 100644
--- a/swig/Capabilities.i
+++ b/swig/Capabilities.i
@@ -8,3 +8,17 @@ by_value_iterator(zypp::Capabilities);
 #ifdef SWIGPERL5
 forwarditer(zypp::Capabilities, zypp::Capability);
 #endif
+
+#ifdef SWIGPYTHON
+%extend  zypp::Capabilities {
+    std::string CapNames()
+    {
+        std::string caps;
+        for (zypp::Capabilities::const_iterator it = self->begin(); it != self->end(); ++it) {
+            std::string tmp = ":" + (*it).asString();
+            caps += tmp;
+        }
+        return caps;
+    }
+}
+#endif
